### PR TITLE
Add shopping cart feature

### PR DESCRIPTION
**Pull request by okucheruk**
This PR add shopping cart feature.

**Highlights:**
- `ShoppingCart` is immutable,
- `ShoppingCartService` is used for CRUD operations on `ShoppingCart`.

### CODE REVIEW

**Comment by okucheruk** [comment]
@Oleh-Stefanovych, PTAL.

File: src/main/java/io/javaclasses/buynbye/ShoppingCartService.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package io.javaclasses.buynbye;
+
+import java.util.ArrayList;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A class which represents shopping cart CRUD operations.
```
Comment (Oleh-Stefanovych) [inline src/main/java/io/javaclasses/buynbye/ShoppingCartService.java]: Looks better `A class which represents CRUD operations for shopping cart items`

File: src/test/java/io/javaclasses/buynbye/ShoppingCartServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class ShoppingCartServiceTest {
+
+    private final ShoppingCartService shoppingCartService;
+    private final ShoppingItem shoppingItem;
+
+    ShoppingCartServiceTest() {
+        this.shoppingCartService = new ShoppingCartService();
+        this.shoppingItem = new ShoppingItem(new Product("Another Brick", "In the wall", 1.0),
```
Comment (Oleh-Stefanovych) [inline src/test/java/io/javaclasses/buynbye/ShoppingCartServiceTest.java]: Please extract `Product` instance.

File: src/main/java/io/javaclasses/buynbye/ShoppingCartService.java (hunk 1 above)
Comment (okucheruk) [reply src/main/java/io/javaclasses/buynbye/ShoppingCartService.java]: replaced.

File: src/test/java/io/javaclasses/buynbye/ShoppingCartServiceTest.java (hunk 2 above)
Comment (okucheruk) [reply src/test/java/io/javaclasses/buynbye/ShoppingCartServiceTest.java]: Extracted.

File: src/main/java/io/javaclasses/buynbye/ShoppingCartService.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    public ShoppingCart removeItem(final ShoppingItem shoppingItem) {
+
+        checkNotNull(shoppingItem);
+
+        return new ShoppingCart(new ArrayList<ShoppingItem>());
+    }
+
+    /**
+     * Increases shopping item quantity;
```
[the remark this answers is not in the document]
Comment (okucheruk) [reply src/main/java/io/javaclasses/buynbye/ShoppingCartService.java]: Fixed.

**Comment by okucheruk** [comment]
@Oleh-Stefanovych, fixed, PTAL.

File: src/main/java/io/javaclasses/buynbye/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    List<ShoppingItem> items() {
+        return Collections.unmodifiableList(this.items);
+    }
+
+    /**
+     * Returns current shopping cart items count.
+     *
+     * @return current shopping cart items count.
+     */
+    int itemsCount() {
```
Comment (Oleh-Stefanovych) [inline src/main/java/io/javaclasses/buynbye/ShoppingCart.java]: Looks redundant. Please, remove.

File: src/test/java/io/javaclasses/buynbye/ShoppingCartServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+        int purchaseItemsCount = shoppingCart.itemsCount();
+        assertEquals(1, purchaseItemsCount,
+                     "Decreasing items quantity should not reduce shopping cart items count.");
+
+        double purchaseTotalCost = shoppingCart.totalProductsCost();
+        assertEquals(695.0, purchaseTotalCost,
+                     "Decreasing items quantity should reduce shopping cart total cost.");
+
+    }
```
Comment (Oleh-Stefanovych) [inline src/test/java/io/javaclasses/buynbye/ShoppingCartServiceTest.java]: Please add tests for negative cases.

File: src/main/java/io/javaclasses/buynbye/ShoppingCart.java (hunk 4 above)
Comment (okucheruk) [reply src/main/java/io/javaclasses/buynbye/ShoppingCart.java]: Removed.

**Comment by okucheruk** [comment]
@Oleh-Stefanovych, fixed, PTAL.
